guide/src/free-threading.md

```diff
@@ -0,0 +1,127 @@
+# Supporting Free-Threaded CPython
+
+CPython 3.13 introduces an experimental build of CPython that does not rely on
+the global interpreter lock for thread safety. As of version 0.23, PyO3 also has
+preliminary support for building rust extensions for the free-threaded Python
+build and support for calling into free-threaded Python from Rust.
+
+The main benefit for supporting free-threaded Python is that it is no longer
+necessary to rely on rust parallelism to achieve concurrent speedups using
+PyO3. Instead, you can parallelise in Python using the
+[`threading`](https://docs.python.org/3/library/threading.html) module, and
+still expect to see see multicore speedups by exploiting threaded concurrency in
+Python, without any need to release the GIL. If you have ever needed to use
+`multiprocessing` to achieve a speedup for some algorithm written in Python,
+free-threading will likely allow the use of Python threads instead for the same
+workflow.
+
+If you want more background on free-threaded Python in general, see the [what's
+new](https://docs.python.org/3.13/whatsnew/3.13.html#whatsnew313-free-threaded-cpython)
+entry in the CPython docs, the [HOWTO
+guide](https://docs.python.org/3.13/howto/free-threading-extensions.html#freethreading-extensions-howto)
+for porting C extensions, and [PEP 703](https://peps.python.org/pep-0703/),
+which provides the technical background for the free-threading implementation in
+CPython.
+
+This document provides advice for porting rust code using PyO3 to run under
+free-threaded Python. While many simple PyO3 uses, like defining an immutable
+python class, will likely work "out of the box", there are currently some
+limitations. 
+
+## Many symbols exposed by PyO3 have `GIL` in the name
+
+We are aware that there are some naming issues in the PyO3 API that make it
+awkward to work in an environment where there is no GIL. We plan to change the
+names of these types to deemphasize the role of the GIL in future versions of
+PyO3, but for now you should remember that the use of the term `GIL` in
+functions and types like `with_gil` and `GILOnceCell` is historical.
+
+Instead, you can think about whether or not you a rust scope has access to a
+Python **thread state** in `ATTACHED` status. See [PEP
+703](https://peps.python.org/pep-0703/#thread-states) for more background about
+Python thread states and status. In order to use the CPython C API in both the
+GIL-enabled and free-threaded builds of CPython, you must own an attached
+Python thread state. The `with_gil` function sets this up and releases the
+thread state after the closure passed to `with_gil` finishes. Similarly, in both
+the GIL-enabled and free-threaded build, you must use `allow_threads` in
+order to use rust threads. Both of `with_gil` and `allow_threads` tell CPython
```

Comment:
I'm not quite sure how to interpret the "Rust threads" part. I don't think there is a problem in just spawning Rust thread an doing something different non Python related in the background. It's more about detaching the _current_ thread from it's interaction with the Interpreter and handing back control, if I got that correctly.